src/designspace.rs

```diff
@@ -78,6 +81,82 @@ pub struct AxisMapping {
     pub output: f32,
 }
 
+/// Describes the substitution [rules] of the Designspace.
+///
+/// [rules]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#rules-element
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct Rules {
+    /// Indicates whether substitution rules should be applied before or after
+    /// other glyph substitution features.
+    #[serde(rename = "@processing")]
+    pub processing: RuleProcessing,
+    /// The rules.
+    #[serde(default, rename = "rule")]
+    pub rules: Vec<Rule>,
+}
+
+/// Indicates whether substitution rules should be applied before or after other
+/// glyph substitution features.
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum RuleProcessing {
+    /// Apply before other substitution features.
+    #[default]
+    First,
+    /// Apply after other substitution features.
+    Last,
+}
+
+/// Describes a single set of substitution rules.
+///
+/// Does not support standalone `<condition>` elements outside a
+/// `<conditionset>`.
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct Rule {
+    /// Name of the rule.
+    #[serde(rename = "@name")]
+    pub name: Option<String>,
```

Comment:
let's use `GlyphName` as the type for these?

Reply:
Hm, the rule name is not a glyph name, but I'll change the substitution names :)

Reply:
oops sorry yea just the glyph names. :)